app/views/user_sessions/_new.html.erb

```diff
@@ -38,4 +38,15 @@
     <p style="color: #888;"><%= raw t('user_sessions.new.reset_by_clicking_here', :url1 => "/reset/") %></p>
   </div>
   <% end %>
+  <% if !current_user %>
+    <script>
+      $(document).ready(function() {
+        $('a.requireLogin').on('click', function(e) {
+          e.preventDefault();
+          $('#loginModal').modal();
```

Comment:
If the modal is toggled like this should I remove the parameters **data-toggle="modal" data-target="#loginModal" data-mode="login"** from elements that toggle the modal?

Reply:
EDIT: I've implemented this code and it does in fact toggle the modal, but it shows both partials as there is no way to tell the modal what partial to render without the data-mode attribute, which is only passed when the button is clicked.

Reply:
I've the same doubt a @JonathanXu1 .. How do we exactly toggle the modal after implementing this code?

Reply:
@oorjitchowdhary  He toggles the modal on line 46, but it would result in both partials (login and signup) being shown on that modal. Example: https://github.com/publiclab/plots2/pull/4209#issuecomment-446346727
I've added changes that fixed this in my PR

Reply:
ok great. So, you have integrated @kevinzluo changes into your pr and also fix the errors.
Great teamwork